c2c: unskip some tests under stress race

msbutler opened the pull request:
Unskipping:
- TestTenantStreamingPauseOnPermanentJobError
- TestTenantStreamingPauseResumeIngestion
- TestTenantStreamingProducerJobTimedOut
- TestTenantStreamingCheckpoint

Remove redundant TestTenantStreaming test

Epic: None

Release note: None

msbutler committed:
c2c: unskip TestTenantStreamingProducerJobTimedOut

This test has been skipped under stress race for reasons I don't understand.
This patch unskips the test and removes a somewhat unnecesssary fingerprint
check that seem unrelated to the test.

Epic: None

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/106265)
<!-- Reviewable:end -->

msbutler committed:
c2c: unskip TestTenantStreamingPauseResumeIngestion under stress and race

Epic: none

Release Note: none

msbutler committed:
c2c: unskip TestTenantStreamingPauseOnPermanentJobError under stress and race

msbutler committed:
c2c: unskip TestTenantStreamingCheckpoint; remove TestTenantStreaming

This patch unskips TestTenantStreamingCheckpoint under stress race and removes
TestTenantStreaming which adds no test coverage beyond TestDataDriven/simple.

Epic: none

Release note: none

lidorcarmel commented on pkg/ccl/streamingccl/streamingest/replication_stream_e2e_test.go:
```diff
@@ -229,12 +216,13 @@ func TestTenantStreamingPauseOnPermanentJobError(t *testing.T) {
 
-	// Make ingestion error out only once.
+	// Make ingestion error out only once to ensure the job conducts one retryable
+	// error. It's fine to close the channel before the receiver gets the error,
+	// as the channel is buffered.
```
and if it wasn't buffered it was still fine..
maybe:
It's fine to close the buffered channel, the receiver will still get the error.
or something like that?

msbutler replied:
Done.

lidorcarmel commented on pkg/ccl/streamingccl/streamingest/replication_stream_e2e_test.go:
```diff
@@ -340,20 +324,13 @@ func TestTenantStreamingCheckpoint(t *testing.T) {
 		require.Less(t, checkpointMinTime.UnixNano(), clientStartTime.GoTime().UnixNano())
 	}
 
-	// After cutover, changes to source won't be streamed into destination cluster.
-	c.SrcExec(func(t *testing.T, sysSQL *sqlutils.SQLRunner, tenantSQL *sqlutils.SQLRunner) {
-		tenantSQL.Exec(t, `INSERT INTO d.t2 VALUES (3);`)
-	})
-	// Check the dst cluster didn't receive the change after a while.
-	<-time.NewTimer(3 * time.Second).C
-	c.RequireDestinationFingerprintAtTimestamp(cutoverFingerprint, c.DestSysServer.Clock().Now().AsOfSystemTime())
```
as much as I don't like the 3 seconds thing.. isn't this an important part of this test?

msbutler replied:
we already exercise this in a different test [here](https://github.com/cockroachdb/cockroach/blob/bad486497ada5cc5465398c71fe8dedb334e59ef/pkg/ccl/streamingccl/streamingest/testdata/simple#L93). I don't think doing this  again in this test adds more coverage.

msbutler commented:
unrelated flake

msbutler commented:
TFTR!

bors r=lidorcarmel

msbutler commented:
blathers backport 23.1
